notion-core/src/test/scala/com/ambiata/notion/core/LocationIOSpec.scala

```diff
@@ -111,4 +117,62 @@ class LocationIOSpec extends Specification with ScalaCheck { def is = s2"""
     _ <- i.writeUtf8(p, data)
     e <- i.exists(p)
   } yield e ==== true)
+
+  def fileFirst = prop { (loc: LocationTemporary, linesNumber: NaturalIntSmall) =>
+    for {
+      ll         <- writeLines(loc, linesNumber)
+      (l, lines) =  ll
+      i          <- loc.io
+      line       <- i.firstLine(l)
+    } yield line must_== lines.headOption
+  }.set(maxSize = 10)
+
+  def fileLast = prop { (loc: LocationTemporary, linesNumber: NaturalIntSmall) =>
+    for {
+      ll         <- writeLines(loc, linesNumber)
+      (l, lines) =  ll
+      i          <- loc.io
+      line       <- i.lastLine(l)
+    } yield line must_== lines.lastOption
+  }.set(maxSize = 10)
+
+  def fileHead = prop { (loc: LocationTemporary, linesNumber: NaturalIntSmall, requestedLinesNumber: NaturalIntSmall) =>
+    for {
+      ll         <- writeLines(loc, linesNumber)
+      (l, lines) =  ll
+      i          <- loc.io
+      head       <- i.head(l, requestedLinesNumber.value)
+    } yield head must_== lines.take(requestedLinesNumber.value)
+  }
+
+  def fileTail = prop { (loc: LocationTemporary, linesNumber: NaturalIntSmall, requestedLinesNumber: NaturalIntSmall) =>
+    for {
+      ll         <- writeLines(loc, linesNumber)
+      (l, lines) =  ll
+      i          <- loc.io
+      tail       <- i.tail(l, requestedLinesNumber.value)
+    } yield tail must_== lines.drop(lines.size - requestedLinesNumber.value)
+  }.set(maxSize = 10)
+
+  def fileTailAndLinesNumber = prop { (loc: LocationTemporary, linesNumber: NaturalIntSmall, requestedLinesNumber: NaturalIntSmall) =>
```

Comment:
Can you just add another test that verifies this gets the same result as mundane readlines on local disk as a cross check.